### DISCUSSION TITLE
Simple Newton solver

### DISCUSSION THREAD

**Discussion by CVerhoosel** [evalf/nutils#690] [Q&A] [answered]
Dear Evalf developers and users,

In quite a few cases we have experienced the standard Newton solver (with default arguments) failing to converge, or to converge very slowly. This happens for example for large strain elasticity with hyperelastic materials in rather complex three-D problems. When we make our own simple Newton solver (without linesearch etc, really just Newton iterations) significantly improved convergence is observed. It is not entirely clear where this comes from, but it may have something to do with the (lack of) smoothness of the problem.

Our question is basically the following: **Is it possible to supply the Nutils Newton solver with arguments such that it behaves as a standard Newton solver, without any extra bells and whistles?** This would be convenient, as we then would not need to implement our own Newton solver.

Best, CV

**Comment by joostvanzwieten** [comment]
You can disable line search by passing `linesearch=lambda *args: (1, True)` to `nutils.solver.newton`. Furthermore, you can control the tolerance of the linear solver using parameters `linatol` and `linrtol`. If you want to use a fixed tolerance for all iterations, pass `linatol=TOL` and `linrtol=0` to `nutils.solver.newton`, where `TOL` is the desired tolerance.

**Reply by gertjanvanzwieten** [reply]
This is the answer, though in this form it probably won't work with caching enabled. Since vanilla Newton is such a standard algorithm I think we'd do well to make it a bit more accessible. That said, I'd be interested to examine the cases that fail to converge. Slow convergence I can understand - the standard algorithm is quite careful to steer downhill, perhaps too careful - but failure I find more difficult to see for precisely the same reason.

**Reply by CVerhoosel** [reply]
Thanks for the answer. We can get you a script where this happens, but these are not dummy examples, as for such cases it does work.

**Comment by gertjanvanzwieten** [comment, marked as answer]
Update: as of 4f5edb48 (merged 2022-12-03) you can simply write `linesearch=None` to switch to vanilla Newton.
